ndcube/asdf/converters/ndcube_converter.py

```diff
@@ -0,0 +1,30 @@
+from asdf.extension import Converter
+import numpy as np
+
+
+class NDCubeConverter(Converter):
+    tags = ["tag:sunpy.org:ndcube/ndcube/NDCube-*"]
+
+    @property
+    def types(self):
+        from ndcube.ndcube import NDCube
+        return [NDCube]
+
+    def select_tag(self, obj, tags, ctx):
```

Comment:
I would leave this out for now. As you've noted in the comments (which are very helpful!) the ordering here can sometimes lead to confusion. The tag versions can be managed by versioning the extension (in different manifests) so that for a given extension there is only one tag version (and we won't need a `select_tag`).